src/server.c

```diff
@@ -605,8 +592,13 @@ int htNeedsResize(dict *dict) {
 /* If the percentage of used slots in the HT reaches HASHTABLE_MIN_FILL
  * we resize the hash table to save memory */
 void tryResizeHashTables(int dbid) {
-    if (htNeedsResize(server.db[dbid].dict))
-        dictResize(server.db[dbid].dict);
+    dict *d;
+    dbIterator dbit;
+    dbInitIterator(&dbit, &server.db[dbid]);
+    while ((d = dbNextDict(&dbit))) {
```

Comment:
It's true that the number of tables increased, but total amount of work needed to resize the dictionary remains the same. The only overhead should be iterating through slots, but it shouldn't take much time to become a concern for cron.

Reply:
i don't understand what work do you refer to.
all the "work" of htNeedsResize, dictResize, and dictExpand is repeated for each slot rather than just being done once.
the only thing that remains roughly the same volume is of zcalloc (many small callocs rather than one that's bigger), so it's true that the amount of memory being zeroed may not change much, but the overhead per allocation call is likely to be large.

Reply:
You are right, there is certainly some overhead, by "work" I meant useful work e.g. actual resizing of dictionaries, as you said amount of memory that should be allocated, zeroed out should be roughly the same. 
I've added a timer around this loop and it takes less than 0.5ms on average to run this loop on the node with 16k slots, I think you right, this might be a bit too much. Let me add upper limit on the number of slots it can process at a time.